libs/service-portal/icelandic-names-registry/src/components/Editor/Editor.tsx

```diff
@@ -150,7 +150,10 @@ const Editor: FC = () => {
       }
 
       try {
-        await deleteName({ variables: { input: { id: body.id } } })
+        await deleteName({
+          variables: { input: { id: body.id } },
+        })
```

Comment:
Would like to see a gif of this, in terms of seeing the update.

There are some implications in this component, most likely due to the fact that the AST is split from the component so it's hard to know what is going on.

The lazy query doesn't rely on the cache, so a full round-trip is needed every time a mutation occurs. Since I don't have a gif I don't know if the query is slow and if it needs to be speed up. By querying the correct fields after mutating an object, the Apollo cache should handle updating its store based on the unique ID that should be fetched after a mutation (e.g. the update name mutation).
Where the store doesn't already have the ID, the store could be updated based on the objects returned from the mutation (e.g. create name mutation). Similar approach can be used for the delete name mutation, along with optimistically updating the store if wanted for better UX.
This would both reduce the round-trips needed for the search query, as well as being able to use the cache.

Reply:
I made some changes to it, I think it should be better now. Now it updates the cache instead of making a full round-trip on mutations.

![image](https://user-images.githubusercontent.com/8450096/117731110-17cafd80-b1dd-11eb-85e5-7c45d4125d71.png)

Reply:
Nice, good job!